lib/filesystem/CZipSaver.cpp

```diff
@@ -82,12 +82,11 @@ si64 CZipOutputStream::write(const ui8 * data, si64 size)
 
 ///CZipSaver
 CZipSaver::CZipSaver(std::shared_ptr<CIOApi> api, const boost::filesystem::path & path):
-	ioApi(api),
+	ioApi(std::move(api)),
 	zipApi(ioApi->getApiStructure()),
-	handle(nullptr),
+	handle(zipOpen2_64(path.c_str(), APPEND_STATUS_CREATE, nullptr, &zipApi)),
```

Comment:
Probably equivalent since old code didn't crash.
But this is pure luck.
zipOpen2_64 requires C string (char*), and not boost::filesystem::path.

Old code worked due to omni-powerful C-style cast but was anything but correct

Reply:
Yes, this is more correct version.